lib/ex_aws/request.ex

```diff
@@ -84,9 +86,11 @@ defmodule ExAws.Request do
     {:attempt, attempt + 1}
   end
 
-  # TODO: make exponential
-  # TODO: add jitter
   def backoff(attempt) do
-    :timer.sleep(attempt * 1000)
+    (@base_backoff_in_ms * :math.pow(2, attempt))
+    |> min(@max_backoff_in_ms)
+    |> trunc
+    |> :rand.uniform
```

Comment:
It's a little unintuitive, but based on the article I believe that's on purpose.

They have an "equal" jitter variant which is closer to what you're describing, they describe it like "The intuition behind this one is that it prevents very short sleeps, always keeping some of the slow down from the backoff."

They also describe a 3rd variant they call "decorated" jitter you could look into.

At any rate, they saw better results with the "full" jitter approach though vs the "equal" jitter approach. (And even the equal jitter approach had more jitter than your example, from around 50% to 150% of the base value).

"Of the jittered approaches, “Equal Jitter” is the loser. It does slightly more work than “Full Jitter”, and takes much longer. The decision between “Decorrelated Jitter” and “Full Jitter” is less clear. The “Full Jitter” approach uses less work, but slightly more time. Both approaches, though, present a substantial decrease in client work and server load."

Like I said though, feel free to play with it, but I'm not sure that guaranteeing a general curve actually matters.